user_tools/src/spark_rapids_pytools/common/sys_storage.py

```diff
@@ -99,6 +99,18 @@ def copy_resource(cls, src: str, dest: str) -> str:
                                    f'Resource {abs_src} does not exist')
         return shutil.copy2(abs_src, abs_dest)
 
+    @classmethod
+    def cache_resource(cls, src: str, dest: str):
+        abs_src = os.path.abspath(src)
+        abs_dest = os.path.abspath(dest)
+        # check if path exists
+        if not os.path.exists(abs_src):
+            raise StorageException('Error copying resource on local disk. '
+                                   f'Resource {abs_src} does not exist')
+        with open(abs_src, 'rb') as s:
+            with open(abs_dest, 'wb') as d:
+                shutil.copyfileobj(s, d)
+
```

Comment:
I kinda get what you are trying to achieve here.
The question is do we need to copy the file from the resource directory into another directory? Or is it because our code assumes that all the cached -files exist in the caching_directory?

Reply:
I copied the file into the cache directory so that it is consistent with other cached files, but we can update the code to avoid copying the file and store the file path with the resource directory in dictionary `cache_files`.